### PR TITLE
[Rakefile] Provide build task to try softwares without building a project

### PR DESCRIPTION
**Pull request by masci**
## What
Added a `rake` task to build a software along with its dependencies.

## Why
Avoid creating a fake project with given dependencies, just clone this repo, make changes, build to see they work.

## Notes
 * I had to merge some stuff from upstream, my changes were added so that further pulls should be smooth
 * Tests are failing because they have to, we need to address failures at some point

### CODE REVIEW

File: config/software/chef.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+name "chef"
```
Comment (masci) [inline config/software/chef.rb]: I don't know why Omnibus wants this to be here, still investigating

File: lib/omnibus-software/version.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module OmnibusSoftware
+  VERSION = "5.0.0"
```
Comment (olivielpeau) [inline lib/omnibus-software/version.rb]: nitpick: the upstream version is `4.0.0`: https://github.com/chef/omnibus-software/blob/c47c8200a9d112db2d73b4d7252bcf9dfd164ca6/lib/omnibus-software/version.rb#L2

we should probably use that one? (or maybe keep the same old `3.0.0` version)

File: config/software/curl.rb [hunk 3]
```diff
@@ -47,7 +47,8 @@
            "--without-gnutls",
            "--without-librtmp",
            "--with-ssl=#{install_dir}/embedded",
-           "--with-zlib=#{install_dir}/embedded"].join(" ")
+           "--with-zlib=#{install_dir}/embedded",
+           "--with-nghttp2=#{install_dir}/embedded"].join(" ")
```
Comment (olivielpeau) [inline config/software/curl.rb]: this probably shouldn't be part of the PR :)

File: omnibus-software.gemspec [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
 
 Gem::Specification.new do |s|
   s.name        = "omnibus-software"
   s.version     = OmnibusSoftware::VERSION
-  s.authors     = ['Chef Software, Inc.']
-  s.email       = ['legal@getchef.com']
-  s.homepage    = "http://github.com/DataDog/omnibus-software"
+  s.authors     = ["Chef Software, Inc."]
+  s.email       = ["legal@getchef.com"]
+  s.homepage    = "http://github.com/opscode/omnibus-software"
```
Comment (olivielpeau) [inline omnibus-software.gemspec]: We should keep `http://github.com/DataDog/omnibus-software` here

**Comment by olivielpeau** [comment]
Looks good apart from my comments, thanks for adding these rake commands 🚀 

Let's fix the CI in a separate PR

File: lib/omnibus-software/version.rb (hunk 2 above)
Comment (masci) [inline lib/omnibus-software/version.rb]: let's use `4.0.0` as it comes from upstream, not sure where it's used but not changing it could be ease further pulls from them
